benches/oneshot/ipa.rs

```diff
@@ -1,23 +1,34 @@
+use clap::Parser;
 use rand::Rng;
 use raw_ipa::error::Error;
 use raw_ipa::ff::Fp32BitPrime;
 use raw_ipa::ipa_test_input;
 use raw_ipa::protocol::ipa::ipa;
 // use raw_ipa::protocol::ipa::{ipa_wip_malicious};
 
-use raw_ipa::protocol::{BreakdownKey, MatchKey};
+use raw_ipa::protocol::{ipa, BreakdownKey, MatchKey};
+use raw_ipa::secret_sharing::SharedValue;
 use raw_ipa::test_fixture::input::GenericReportTestInput;
 use raw_ipa::test_fixture::{Runner, TestWorld, TestWorldConfig};
 use std::num::NonZeroUsize;
 use std::time::Instant;
 
+#[derive(Debug, Parser)]
+pub struct IpaBenchArgs {
+    #[arg(long, help = "bits of match key to use in IPA circuit.")]
+    pub match_key_bits: Option<u8>,
```

Comment:
There are a lot of other tuning parameters in this file that might benefit from similar treatment.  See BATCHSIZE in particular.  Also, your Query struct is full of goodies that might benefit from slapping clap annotations on it.

You can (and maybe should) use the collapsing mode (`#[flatten]`, from memory) and just include the IPA options as a sub-element of this.  More comments below.